contributor-docs/adrs/adr-005.md

```diff
@@ -0,0 +1,65 @@
+# ADR 5: Migration plan for deprecated components
+
+## Status
+
+Proposed
+
+## Context
+
+As we work on maturity of our components, we will sometimes need to deprecate components that are replaced with new components.
+
+## Here are the 3 proposed stages:
+
+### Stage 1
+
+Start new component outside the main bundle. Folks who want to try it, need to explicitly import it from the `drafts` bundle.
+
+`import { ActionMenu} from '@primer/react/drafts'`
```

Comment:
sidenote: thinking about these stages, I started thinking about the name again. 

In this context, I really like the name `'@primer/react/labs'` but that's a tiny detail

Reply:
I like `/labs` too ❤️

Reply:
While I like labs too, would this make sense for users to import from `labs` as it feels more like a function. Would something like `import { ActionMenu} from '@primer/react/experimental'` make more sense? (I feel like I'm having deja vu, has this already been proposed? 😅 )

Might also be worth syncing with ViewComponents on this naming convention for ecosystem consistency. I saw a PVC thread where they were thinking about calling it `experimental`. Will try find it.

Reply:
I can imagine that automating migrations (to maintain the same functionality without upgrading anything) when you make the breaking change would be really messy if a file has other `@primer/react` or `@primer/react/drafts` imports.

Is performance the main reason to bundle drafts separately? I'd have thought that with tree-shaking that wasn't much of a concern but idk how our js is all set up, and given we want to include the draft components in the docs we're not exactly trying to hide them.

If they were all in the main bundle you could have `ActionMenu` and `ActionMenuDraft`/`ActionMenuNext`, then renaming them to `ActionMenuDeprecated` and `ActionMenu` when you make the breaking change would be trivial. Just a thought.

Reply:
Hi! 

Sorry for chiming in late! I don't think automating migrations is a goal here. Most of the components in drafts right now have breaking API changes, so optimising for import paths isn't a goal here. 

> Is performance the main reason to bundle drafts separately?

Separation of tracks was the main reason here. Seeing `ActionMenu` and `ActionMenuNext` can be confusing to choose from. (We did not do ourselves any favours in the status page with v1 and v2).

The correct way to title them in the docs might be `ActionMenu` and `Experimental/ActionMenu`. I updated the PR description and ADR to add the usage of parallel tracks

![Parallel tracks](https://user-images.githubusercontent.com/1863771/150127636-1c7332d5-cd36-4944-ab6b-3bfc18f977f6.png)